pkg/coordinator/s3coordinator/coordinator_s3.go

```diff
@@ -351,18 +352,26 @@ func (c *CoordinatorS3) FinishOperation(operationID string, shardIndex int, task
 
 // Utility functions to interact with S3.
 func (c *CoordinatorS3) putObject(key string, body []byte) error {
+	fullKey := key
+	if c.path != "" {
+		fullKey = c.path + "/" + key
+	}
```

Comment:
this 4 lines repeats 3 times, worth to make a small helper method `c.fullKey(key)`